File: gaia-sdk-python/gaia_sdk/graphql/request/input/CreateSkillImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateSkillImpulse():
    """
    The specification to create a skill instance
    """
    tenantId: str
    qualifier: str
    appendent: str
    labelList: List[str]
    repositoryUri: str

    def __init__(self, tenantId: str, qualifier: str, appendent: str, labelList: List[str], repositoryUri: str):
        self.tenantId = tenantId
        self.qualifier = qualifier
        self.appendent = appendent
        self.labelList = labelList
        self.repositoryUri = repositoryUri

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.qualifier == other.qualifier and self.appendent == other.appendent and self.labelList == other.labelList and self.repositoryUri == other.repositoryUri
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'qualifier': self.qualifier, 'appendent': self.appendent, 'labelList': self.labelList, 'repositoryUri': self.repositoryUri}

File: gaia-sdk-javascript/src/graphql/request/input/CreateSkillImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to create a skill instance
 */
export class CreateSkillImpulse {
public _typeName = "CreateSkillImpulse";
    private tenantId:Uuid;
    private qualifier:String;
    private appendent:String;
    private labelList:Array<String>;
    private repositoryUri:String;
    private repositoryType:String;

    constructor (tenantId:Uuid, qualifier:String, appendent:String, labelList:Array<String>, repositoryUri:String, repositoryType:String) {
        this.tenantId = tenantId;
        this.qualifier = qualifier;
        this.appendent = appendent;
        this.labelList = labelList;
        this.repositoryUri = repositoryUri;
        this.repositoryType = repositoryType;
    }
}
